client/src/common/components/RequireAuth.tsx

```diff
@@ -4,6 +4,7 @@ import { checkValidToken } from '../authorizationSlice';
 import { useAppSelector, useAppDispatch } from '../../app/hook';
 import { getUsers } from '../usersSlice';
 import { getProfile } from '../myProfileSlice';
+import Header from './Header';
 
 const RequireAuth = ({ children }: { children: JSX.Element }) => {
```

Comment:
I searched some about defining `children` and `React.ReactNode` also looks ok to be defined here. But the children for RequireAuth will always be a React element (component) so it's ok we define it as `JSX.element`?
Or this can just be `const RequireAuth: React.FC = ({ children }) => {` If we don't define `children` explicitly?
I'm not sure 🤯

Reply:
I don't really know the difference between `JSX.element` and `React.ReactNode` to be honest.
I tried to render the elements with `const RequireAuth: React.FC = ({ children }: { children: React.ReactNode })` but it returns the error message saying `Its return type 'ReactNode' is not a valid JSX element` and didn't work well.

Also, I tried this: `const RequireAuth: React.FC = ({ children })` but it means we don't specify the type of "children" and "children" could be _"any"_.  It gave me a squiggly line too.

That shows we have to specify the type of children all the time.

I found an alternative way.
You can define props' type like this:
```js
interface IProps {
	children: JSX.Element;
}
```
and set this props' type as the following, it works the same.
```js
const RequireAuth: React.FunctionComponent<IProps> = ({ children })
```